File: app/src/main/java/com/example/infoasia/Dao.java
package com.example.infoasia;

import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

@androidx.room.Dao
public interface Dao {
    @Insert
    public void addCountry(Country country);

    @Update
    public void update(Country country);

    @Query("select * from countries")
    public List<Country> getCountries();

    @Delete
    public void deleteCountry(Country country);
}

File: app/src/main/java/com/example/asiainfo/Dao.kt
package com.example.asiainfo

import androidx.room.*
import androidx.room.Dao

@Dao
interface Dao {
    @Insert
    fun addCountry(country: Country?)

    @Update
    fun update(country: Country?)

    @Query("select * from countries")
    fun getCountries(): List<Country?>?

    @Delete
    fun deleteCountry(country: Country?)
}